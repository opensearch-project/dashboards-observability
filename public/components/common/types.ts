/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export interface FieldCapAttributes {
  type: string;
  searchable: boolean;
  aggregatable: boolean;
}

export interface FieldCapResponse {
  indices: string[];
  fields: Record<string, Record<string, FieldCapAttributes>>;
}
