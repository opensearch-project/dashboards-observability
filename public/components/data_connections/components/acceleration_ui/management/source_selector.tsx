/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { EuiComboBox, EuiFormRow, EuiSpacer, EuiText } from '@elastic/eui';
import React, { useState } from 'react';
import { useEffect } from 'react';

interface DataSourceTypes {
  label: string;
}

export const AccelerationDataSourceSelector = () => {
  const [dataConnections, setDataConnections] = useState<DataSourceTypes[]>([]);
  const [selectedDataConnection, setSelectedDataConnection] = useState<DataSourceTypes[]>([]);
  const [tables, setTables] = useState<DataSourceTypes[]>([]);
  const [selectedTable, setSelectedTable] = useState<DataSourceTypes[]>([]);

  useEffect(() => {
    setDataConnections([
      {
        label: 'spark1',
      },
      {
        label: 'spark2',
      },
    ]);
  }, []);

  useEffect(() => {
    setTables([
      {
        label: 'Table1',
      },
      {
        label: 'Table1',
      },
    ]);
  }, [dataConnections]);

  const onChangeDataConnection = (
    dataConnectionOptions: React.SetStateAction<DataSourceTypes[]>
  ) => {
    setSelectedDataConnection(dataConnectionOptions);
  };
  const onChangeTable = (tableOptions: React.SetStateAction<DataSourceTypes[]>) => {
    setSelectedTable(tableOptions);
  };

  return (
    <>
      <EuiText data-test-subj="datasource-selector-header">
        <h3>Select data connection</h3>
      </EuiText>
      <EuiSpacer size="s" />
      <EuiText size="s" color="subdued">
        Select data connection where the data you want to accelerate resides.{' '}
      </EuiText>
      <EuiSpacer size="s" />
      <EuiFormRow
        label="Data connection"
        helpText="A data connection has to be configured and active to be able to select it and index data from."
      >
        <EuiComboBox
          placeholder="Spark connection name"
          singleSelection={{ asPlainText: true }}
          options={dataConnections}
          selectedOptions={selectedDataConnection}
          onChange={onChangeDataConnection}
        />
      </EuiFormRow>
      <EuiFormRow
        label="Select Table"
        helpText="Select the Spark table that has the data you would like to index."
      >
        <EuiComboBox
          placeholder="Table name"
          singleSelection={{ asPlainText: true }}
          options={tables}
          selectedOptions={selectedTable}
          onChange={onChangeTable}
        />
      </EuiFormRow>
      <EuiSpacer size="xxl" />
    </>
  );
};
