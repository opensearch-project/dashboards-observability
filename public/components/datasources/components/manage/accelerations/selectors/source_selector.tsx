/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { EuiComboBox, EuiComboBoxOptionOption, EuiFormRow, EuiSpacer, EuiText } from '@elastic/eui';
import producer from 'immer';
import React, { useEffect, useState } from 'react';
// import { executeAsyncQuery } from '../../../../common/utils/async_query_helpers';
import { CoreStart } from '../../../../../../../../../src/core/public';
import { CreateAccelerationForm } from '../../../../../../../common/types/data_connections';
import { useToast } from '../../../../../common/toast';
import { hasError, validateDataSource } from '../create/utils';

interface AccelerationDataSourceSelectorProps {
  http: CoreStart['http'];
  accelerationFormData: CreateAccelerationForm;
  setAccelerationFormData: React.Dispatch<React.SetStateAction<CreateAccelerationForm>>;
  selectedDatasource: EuiComboBoxOptionOption[];
}

export const AccelerationDataSourceSelector = ({
  http,
  accelerationFormData,
  setAccelerationFormData,
  selectedDatasource,
}: AccelerationDataSourceSelectorProps) => {
  const { setToast } = useToast();
  const [dataConnections, setDataConnections] = useState<Array<EuiComboBoxOptionOption<string>>>(
    []
  );
  const [selectedDataConnection, setSelectedDataConnection] = useState<
    Array<EuiComboBoxOptionOption<string>>
  >(selectedDatasource.length > 0 ? [{ label: selectedDatasource[0].label }] : []);
  const [databases, _setDatabases] = useState<Array<EuiComboBoxOptionOption<string>>>([]);
  const [selectedDatabase, setSelectedDatabase] = useState<Array<EuiComboBoxOptionOption<string>>>(
    []
  );
  const [tables, _setTables] = useState<Array<EuiComboBoxOptionOption<string>>>([]);
  const [selectedTable, setSelectedTable] = useState<Array<EuiComboBoxOptionOption<string>>>([]);
  const [loadingComboBoxes, setLoadingComboBoxes] = useState({
    dataSource: false,
    database: false,
    dataTable: false,
  });

  const loadDataSource = () => {
    setLoadingComboBoxes({ ...loadingComboBoxes, dataSource: true });
    http
      .get(`/api/get_datasources`)
      .then((res) => {
        const data = res.data.resp;
        setDataConnections(
          data
            .filter((connection: any) => connection.connector.toUpperCase() === 'S3GLUE')
            .map((connection: any) => ({ label: connection.name }))
        );
      })
      .catch((err) => {
        console.error(err);
        setToast(`ERROR: failed to load datasources`, 'danger');
      });
    setLoadingComboBoxes({ ...loadingComboBoxes, dataSource: false });
  };

  const loadDatabases = () => {
    // setLoadingComboBoxes({ ...loadingComboBoxes, database: true });
    // const query = {
    //   lang: 'sql',
    //   query: `SHOW SCHEMAS IN \`${accelerationFormData.dataSource}\``,
    //   datasource: accelerationFormData.dataSource,
    // };
    // executeAsyncQuery(
    //   accelerationFormData.dataSource,
    //   query,
    //   (response: AsyncApiResponse) => {
    //     const status = response.data.resp.status.toLowerCase();
    //     if (status === AsyncQueryStatus.Success) {
    //       let databaseOptions: Array<EuiComboBoxOptionOption<string>> = [];
    //       if (response.data.resp.datarows.length > 0)
    //         databaseOptions = response.data.resp.datarows.map((subArray: any[]) => ({
    //           label: subArray[0],
    //         }));
    //       setDatabases(databaseOptions);
    //       setLoadingComboBoxes({ ...loadingComboBoxes, database: false });
    //     }
    //     if (status === AsyncQueryStatus.Failed || status === AsyncQueryStatus.Cancelled) {
    //       setLoadingComboBoxes({ ...loadingComboBoxes, database: false });
    //     }
    //   },
    //   () => setLoadingComboBoxes({ ...loadingComboBoxes, database: false })
    // );
  };

  const loadTables = () => {
    // setLoadingComboBoxes({ ...loadingComboBoxes, dataTable: true });
    // const query = {
    //   lang: 'sql',
    //   query: `SHOW TABLES IN \`${accelerationFormData.dataSource}\`.\`${accelerationFormData.database}\``,
    //   datasource: accelerationFormData.dataSource,
    // };
    // executeAsyncQuery(
    //   accelerationFormData.dataSource,
    //   query,
    //   (response: AsyncApiResponse) => {
    //     const status = response.data.resp.status.toLowerCase();
    //     if (status === AsyncQueryStatus.Success) {
    //       let dataTableOptions: Array<EuiComboBoxOptionOption<string>> = [];
    //       if (response.data.resp.datarows.length > 0)
    //         dataTableOptions = response.data.resp.datarows.map((subArray) => ({
    //           label: subArray[1],
    //         }));
    //       setTables(dataTableOptions);
    //       setLoadingComboBoxes({ ...loadingComboBoxes, dataTable: false });
    //     }
    //     if (status === AsyncQueryStatus.Failed || status === AsyncQueryStatus.Cancelled) {
    //       setLoadingComboBoxes({ ...loadingComboBoxes, dataTable: false });
    //     }
    //   },
    //   () => setLoadingComboBoxes({ ...loadingComboBoxes, dataTable: false })
    // );
  };

  useEffect(() => {
    loadDataSource();
  }, []);

  useEffect(() => {
    if (accelerationFormData.dataSource !== '') {
      loadDatabases();
    }
  }, [accelerationFormData.dataSource]);

  useEffect(() => {
    if (accelerationFormData.database !== '') {
      loadTables();
    }
  }, [accelerationFormData.database]);

  return (
    <>
      <EuiText data-test-subj="datasource-selector-header">
        <h3>Select data source</h3>
      </EuiText>
      <EuiSpacer size="s" />
      <EuiText size="s" color="subdued">
        Select the data source to accelerate data from. External data sources may take time to load.
      </EuiText>
      <EuiSpacer size="s" />
      <EuiFormRow
        label="Data source"
        helpText="A data source has to be configured and active to be able to select it and index data from."
        isInvalid={hasError(accelerationFormData.formErrors, 'dataSourceError')}
        error={accelerationFormData.formErrors.dataSourceError}
      >
        <EuiComboBox
          placeholder="Select a data source"
          singleSelection={{ asPlainText: true }}
          options={dataConnections}
          selectedOptions={selectedDataConnection}
          onChange={(dataConnectionOptions) => {
            if (dataConnectionOptions.length > 0) {
              setAccelerationFormData(
                producer((accData) => {
                  accData.dataSource = dataConnectionOptions[0].label;
                  accData.formErrors.dataSourceError = validateDataSource(
                    dataConnectionOptions[0].label
                  );
                })
              );
              setSelectedDataConnection(dataConnectionOptions);
            }
          }}
          isClearable={false}
          isInvalid={hasError(accelerationFormData.formErrors, 'dataSourceError')}
          isLoading={loadingComboBoxes.dataSource}
        />
      </EuiFormRow>
      <EuiFormRow
        label="Database"
        helpText="Select the database that contains the tables you'd like to use."
        isInvalid={hasError(accelerationFormData.formErrors, 'databaseError')}
        error={accelerationFormData.formErrors.databaseError}
      >
        <EuiComboBox
          placeholder="Select a database"
          singleSelection={{ asPlainText: true }}
          options={databases}
          selectedOptions={selectedDatabase}
          onChange={(databaseOptions) => {
            if (databaseOptions.length > 0) {
              setAccelerationFormData(
                producer((accData) => {
                  accData.database = databaseOptions[0].label;
                  accData.formErrors.databaseError = validateDataSource(databaseOptions[0].label);
                })
              );
              setSelectedDatabase(databaseOptions);
            }
          }}
          isClearable={false}
          isInvalid={hasError(accelerationFormData.formErrors, 'databaseError')}
          isLoading={loadingComboBoxes.database}
        />
      </EuiFormRow>
      <EuiFormRow
        label="Table"
        helpText="Select the Spark table that has the data you would like to index."
        isInvalid={hasError(accelerationFormData.formErrors, 'dataTableError')}
        error={accelerationFormData.formErrors.dataTableError}
      >
        <EuiComboBox
          placeholder="Select a table"
          singleSelection={{ asPlainText: true }}
          options={tables}
          selectedOptions={selectedTable}
          onChange={(tableOptions) => {
            if (tableOptions.length > 0) {
              setAccelerationFormData(
                producer((accData) => {
                  accData.dataTable = tableOptions[0].label;
                  accData.formErrors.dataTableError = validateDataSource(tableOptions[0].label);
                })
              );
              setSelectedTable(tableOptions);
            }
          }}
          isClearable={false}
          isInvalid={hasError(accelerationFormData.formErrors, 'dataTableError')}
          isLoading={loadingComboBoxes.dataTable}
        />
      </EuiFormRow>
    </>
  );
};
