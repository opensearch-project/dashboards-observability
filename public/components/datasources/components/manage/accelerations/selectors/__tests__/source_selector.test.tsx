/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { waitFor } from '@testing-library/dom';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import React from 'react';
import { coreMock } from '../../../../../../../../../../src/core/public/mocks';
import { CreateAccelerationForm } from '../../../../../../../../common/types/data_connections';
import {
  createAccelerationEmptyDataMock,
  mockDatasourcesQuery,
} from '../../../../../../../../test/accelerations';
import { AccelerationDataSourceSelector } from '../source_selector';

const coreStartMock = coreMock.createStart();

describe('Source selector components', () => {
  configure({ adapter: new Adapter() });

  it('renders source selector with default options', async () => {
    const accelerationFormData = createAccelerationEmptyDataMock;
    const selectedDatasource = 'my_glue';
    const setAccelerationFormData = jest.fn();
    const client = coreStartMock.http;
    client.get = jest.fn().mockResolvedValue(mockDatasourcesQuery);

    const wrapper = mount(
      <AccelerationDataSourceSelector
        http={client}
        selectedDatasource={selectedDatasource}
        accelerationFormData={accelerationFormData}
        setAccelerationFormData={setAccelerationFormData}
      />
    );
    wrapper.update();
    await waitFor(() => {
      expect(
        toJson(wrapper, {
          noKey: false,
          mode: 'deep',
        })
      ).toMatchSnapshot();
    });
  });

  it('renders source selector with different options', async () => {
    const selectedDatasource = 'ds';
    const accelerationFormData: CreateAccelerationForm = {
      ...createAccelerationEmptyDataMock,
      dataSource: 'ds',
      database: 'db',
      dataTable: 'tb',
    };
    const setAccelerationFormData = jest.fn();
    const client = coreStartMock.http;
    client.get = jest.fn().mockResolvedValue(mockDatasourcesQuery);
    client.post = jest.fn().mockResolvedValue([]);
    const wrapper = mount(
      <AccelerationDataSourceSelector
        selectedDatasource={selectedDatasource}
        http={client}
        accelerationFormData={accelerationFormData}
        setAccelerationFormData={setAccelerationFormData}
      />
    );
    wrapper.update();
    await waitFor(() => {
      expect(
        toJson(wrapper, {
          noKey: false,
          mode: 'deep',
        })
      ).toMatchSnapshot();
    });
  });
});
