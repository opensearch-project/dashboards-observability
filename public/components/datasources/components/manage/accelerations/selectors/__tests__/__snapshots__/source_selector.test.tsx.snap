// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Source selector components renders source selector with default options 1`] = `
Array [
  <div
    className="euiText euiText--medium"
    data-test-subj="datasource-selector-header"
  >
    <h3>
      Select data source
    </h3>
  </div>,
  <div
    className="euiSpacer euiSpacer--s"
  />,
  <div
    className="euiText euiText--small"
  >
    <div
      className="euiTextColor euiTextColor--subdued"
    >
      Select the data source to accelerate data from. External data sources may take time to load.
    </div>
  </div>,
  <div
    className="euiSpacer euiSpacer--s"
  />,
  <div
    className="euiFormRow"
    id="some_html_id-row"
  >
    <div
      className="euiFormRow__labelWrapper"
    >
      <label
        aria-invalid={false}
        className="euiFormLabel euiFormRow__label"
        htmlFor="some_html_id"
      >
        Data source
      </label>
    </div>
    <div
      className="euiFormRow__fieldWrapper"
    >
      <div
        aria-describedby="some_html_id-help-0"
        aria-expanded={false}
        aria-haspopup="listbox"
        className="euiComboBox"
        onKeyDown={[Function]}
        role="combobox"
      >
        <div
          className="euiFormControlLayout"
        >
          <div
            className="euiFormControlLayout__childrenWrapper"
          >
            <div
              className="euiComboBox__inputWrap euiComboBox__inputWrap--noWrap"
              data-test-subj="comboBoxInput"
              onClick={[Function]}
              tabIndex={-1}
            >
              <p
                className="euiComboBoxPlaceholder"
              >
                Select a data source
              </p>
              <div
                className="euiComboBox__input"
                style={
                  Object {
                    "display": "inline-block",
                    "fontSize": 14,
                  }
                }
              >
                <input
                  aria-controls=""
                  data-test-subj="comboBoxSearchInput"
                  id="some_html_id"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  role="textbox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "2px",
                    }
                  }
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                />
              </div>
            </div>
            <div
              className="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
            >
              <button
                aria-label="Open list of options"
                className="euiFormControlLayoutCustomIcon euiFormControlLayoutCustomIcon--clickable"
                data-test-subj="comboBoxToggleListButton"
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
                  focusable="false"
                  height={16}
                  role="img"
                  style={null}
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="euiFormHelpText euiFormRow__text"
        id="some_html_id-help-0"
      >
        A data source has to be configured and active to be able to select it and index data from.
      </div>
    </div>
  </div>,
  <div
    className="euiFormRow"
    id="some_html_id-row"
  >
    <div
      className="euiFormRow__labelWrapper"
    >
      <label
        aria-invalid={false}
        className="euiFormLabel euiFormRow__label"
        htmlFor="some_html_id"
      >
        Database
      </label>
    </div>
    <div
      className="euiFormRow__fieldWrapper"
    >
      <div
        aria-describedby="some_html_id-help-0"
        aria-expanded={false}
        aria-haspopup="listbox"
        className="euiComboBox"
        onKeyDown={[Function]}
        role="combobox"
      >
        <div
          className="euiFormControlLayout"
        >
          <div
            className="euiFormControlLayout__childrenWrapper"
          >
            <div
              className="euiComboBox__inputWrap euiComboBox__inputWrap--noWrap"
              data-test-subj="comboBoxInput"
              onClick={[Function]}
              tabIndex={-1}
            >
              <p
                className="euiComboBoxPlaceholder"
              >
                Select a database
              </p>
              <div
                className="euiComboBox__input"
                style={
                  Object {
                    "display": "inline-block",
                    "fontSize": 14,
                  }
                }
              >
                <input
                  aria-controls=""
                  data-test-subj="comboBoxSearchInput"
                  id="some_html_id"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  role="textbox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "2px",
                    }
                  }
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                />
              </div>
            </div>
            <div
              className="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
            >
              <button
                aria-label="Open list of options"
                className="euiFormControlLayoutCustomIcon euiFormControlLayoutCustomIcon--clickable"
                data-test-subj="comboBoxToggleListButton"
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
                  focusable="false"
                  height={16}
                  role="img"
                  style={null}
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="euiFormHelpText euiFormRow__text"
        id="some_html_id-help-0"
      >
        Select the database that contains the tables you'd like to use.
      </div>
    </div>
  </div>,
  <div
    className="euiFormRow"
    id="some_html_id-row"
  >
    <div
      className="euiFormRow__labelWrapper"
    >
      <label
        aria-invalid={false}
        className="euiFormLabel euiFormRow__label"
        htmlFor="some_html_id"
      >
        Table
      </label>
    </div>
    <div
      className="euiFormRow__fieldWrapper"
    >
      <div
        aria-describedby="some_html_id-help-0"
        aria-expanded={false}
        aria-haspopup="listbox"
        className="euiComboBox"
        onKeyDown={[Function]}
        role="combobox"
      >
        <div
          className="euiFormControlLayout"
        >
          <div
            className="euiFormControlLayout__childrenWrapper"
          >
            <div
              className="euiComboBox__inputWrap euiComboBox__inputWrap--noWrap"
              data-test-subj="comboBoxInput"
              onClick={[Function]}
              tabIndex={-1}
            >
              <p
                className="euiComboBoxPlaceholder"
              >
                Select a table
              </p>
              <div
                className="euiComboBox__input"
                style={
                  Object {
                    "display": "inline-block",
                    "fontSize": 14,
                  }
                }
              >
                <input
                  aria-controls=""
                  data-test-subj="comboBoxSearchInput"
                  id="some_html_id"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  role="textbox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "2px",
                    }
                  }
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                />
              </div>
            </div>
            <div
              className="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
            >
              <button
                aria-label="Open list of options"
                className="euiFormControlLayoutCustomIcon euiFormControlLayoutCustomIcon--clickable"
                data-test-subj="comboBoxToggleListButton"
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
                  focusable="false"
                  height={16}
                  role="img"
                  style={null}
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="euiFormHelpText euiFormRow__text"
        id="some_html_id-help-0"
      >
        Select the Spark table that has the data you would like to index.
      </div>
    </div>
  </div>,
]
`;

exports[`Source selector components renders source selector with different options 1`] = `
Array [
  <div
    className="euiText euiText--medium"
    data-test-subj="datasource-selector-header"
  >
    <h3>
      Select data source
    </h3>
  </div>,
  <div
    className="euiSpacer euiSpacer--s"
  />,
  <div
    className="euiText euiText--small"
  >
    <div
      className="euiTextColor euiTextColor--subdued"
    >
      Select the data source to accelerate data from. External data sources may take time to load.
    </div>
  </div>,
  <div
    className="euiSpacer euiSpacer--s"
  />,
  <div
    className="euiFormRow"
    id="some_html_id-row"
  >
    <div
      className="euiFormRow__labelWrapper"
    >
      <label
        aria-invalid={false}
        className="euiFormLabel euiFormRow__label"
        htmlFor="some_html_id"
      >
        Data source
      </label>
    </div>
    <div
      className="euiFormRow__fieldWrapper"
    >
      <div
        aria-describedby="some_html_id-help-0"
        aria-expanded={false}
        aria-haspopup="listbox"
        className="euiComboBox"
        onKeyDown={[Function]}
        role="combobox"
      >
        <div
          className="euiFormControlLayout"
        >
          <div
            className="euiFormControlLayout__childrenWrapper"
          >
            <div
              className="euiComboBox__inputWrap euiComboBox__inputWrap--noWrap"
              data-test-subj="comboBoxInput"
              onClick={[Function]}
              tabIndex={-1}
            >
              <span
                className="euiComboBoxPill euiComboBoxPill--plainText"
              >
                ds
              </span>
              <div
                className="euiComboBox__input"
                style={
                  Object {
                    "display": "inline-block",
                    "fontSize": 14,
                  }
                }
              >
                <input
                  aria-controls=""
                  data-test-subj="comboBoxSearchInput"
                  id="some_html_id"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  role="textbox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "2px",
                    }
                  }
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                />
              </div>
            </div>
            <div
              className="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
            >
              <button
                aria-label="Open list of options"
                className="euiFormControlLayoutCustomIcon euiFormControlLayoutCustomIcon--clickable"
                data-test-subj="comboBoxToggleListButton"
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="euiIcon euiIcon--medium euiFormControlLayoutCustomIcon__icon"
                  focusable="false"
                  height={16}
                  role="img"
                  style={null}
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M13.069 5.157 8.384 9.768a.546.546 0 0 1-.768 0L2.93 5.158a.552.552 0 0 0-.771 0 .53.53 0 0 0 0 .759l4.684 4.61c.641.631 1.672.63 2.312 0l4.684-4.61a.53.53 0 0 0 0-.76.552.552 0 0 0-.771 0Z"
                    fillRule="non-zero"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="euiFormHelpText euiFormRow__text"
        id="some_html_id-help-0"
      >
        A data source has to be configured and active to be able to select it and index data from.
      </div>
    </div>
  </div>,
  <div
    className="euiFormRow"
    id="some_html_id-row"
  >
    <div
      className="euiFormRow__labelWrapper"
    >
      <label
        aria-invalid={false}
        className="euiFormLabel euiFormRow__label"
        htmlFor="some_html_id"
      >
        Database
      </label>
    </div>
    <div
      className="euiFormRow__fieldWrapper"
    >
      <div
        aria-describedby="some_html_id-help-0"
        aria-expanded={false}
        aria-haspopup="listbox"
        className="euiComboBox"
        onKeyDown={[Function]}
        role="combobox"
      >
        <div
          className="euiFormControlLayout"
        >
          <div
            className="euiFormControlLayout__childrenWrapper"
          >
            <div
              className="euiComboBox__inputWrap euiComboBox__inputWrap--noWrap"
              data-test-subj="comboBoxInput"
              onClick={[Function]}
              tabIndex={-1}
            >
              <p
                className="euiComboBoxPlaceholder"
              >
                Select a database
              </p>
              <div
                className="euiComboBox__input"
                style={
                  Object {
                    "display": "inline-block",
                    "fontSize": 14,
                  }
                }
              >
                <input
                  aria-controls=""
                  data-test-subj="comboBoxSearchInput"
                  id="some_html_id"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  role="textbox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "2px",
                    }
                  }
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                />
              </div>
            </div>
            <div
              className="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
            >
              <button
                aria-label="Open list of options"
                className="euiFormControlLayoutCustomIcon euiFormControlLayoutCustomIcon--clickable"
                data-test-subj="comboBoxToggleListButton"
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="euiIcon euiIcon--medium euiFormControlLayoutCustomIcon__icon"
                  focusable="false"
                  height={16}
                  role="img"
                  style={null}
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M13.069 5.157 8.384 9.768a.546.546 0 0 1-.768 0L2.93 5.158a.552.552 0 0 0-.771 0 .53.53 0 0 0 0 .759l4.684 4.61c.641.631 1.672.63 2.312 0l4.684-4.61a.53.53 0 0 0 0-.76.552.552 0 0 0-.771 0Z"
                    fillRule="non-zero"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="euiFormHelpText euiFormRow__text"
        id="some_html_id-help-0"
      >
        Select the database that contains the tables you'd like to use.
      </div>
    </div>
  </div>,
  <div
    className="euiFormRow"
    id="some_html_id-row"
  >
    <div
      className="euiFormRow__labelWrapper"
    >
      <label
        aria-invalid={false}
        className="euiFormLabel euiFormRow__label"
        htmlFor="some_html_id"
      >
        Table
      </label>
    </div>
    <div
      className="euiFormRow__fieldWrapper"
    >
      <div
        aria-describedby="some_html_id-help-0"
        aria-expanded={false}
        aria-haspopup="listbox"
        className="euiComboBox"
        onKeyDown={[Function]}
        role="combobox"
      >
        <div
          className="euiFormControlLayout"
        >
          <div
            className="euiFormControlLayout__childrenWrapper"
          >
            <div
              className="euiComboBox__inputWrap euiComboBox__inputWrap--noWrap euiComboBox__inputWrap-isLoading"
              data-test-subj="comboBoxInput"
              onClick={[Function]}
              tabIndex={-1}
            >
              <p
                className="euiComboBoxPlaceholder"
              >
                Select a table
              </p>
              <div
                className="euiComboBox__input"
                style={
                  Object {
                    "display": "inline-block",
                    "fontSize": 14,
                  }
                }
              >
                <input
                  aria-controls=""
                  data-test-subj="comboBoxSearchInput"
                  id="some_html_id"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  role="textbox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "2px",
                    }
                  }
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                />
              </div>
            </div>
            <div
              className="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
            >
              <span
                className="euiLoadingSpinner euiLoadingSpinner--medium"
              />
              <button
                aria-label="Open list of options"
                className="euiFormControlLayoutCustomIcon euiFormControlLayoutCustomIcon--clickable"
                data-test-subj="comboBoxToggleListButton"
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="euiIcon euiIcon--medium euiFormControlLayoutCustomIcon__icon"
                  focusable="false"
                  height={16}
                  role="img"
                  style={null}
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M13.069 5.157 8.384 9.768a.546.546 0 0 1-.768 0L2.93 5.158a.552.552 0 0 0-.771 0 .53.53 0 0 0 0 .759l4.684 4.61c.641.631 1.672.63 2.312 0l4.684-4.61a.53.53 0 0 0 0-.76.552.552 0 0 0-.771 0Z"
                    fillRule="non-zero"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="euiFormHelpText euiFormRow__text"
        id="some_html_id-help-0"
      >
        Select the Spark table that has the data you would like to index.
      </div>
    </div>
  </div>,
]
`;
